### PR TITLE
Optimize Uniform grid

### PR DESCRIPTION
**Pull request by Jelled1st**
This update brings updating the uniform grid instead of reconstruction of the uniform grid as before.

### CODE REVIEW

File: legion/engine/physics/broadphasecollisionalgorithms/broadphaseuniformgrid.cpp [hunk 1]
```diff
@@ -1,11 +1,170 @@
...
+            //    cellIndices.erase(cell);
+            //    toRemove.push_back(index);
+            //    log::debug("Removing: {}", cell);
+            //}
+            //for (auto index : pair_range<std::set<size_t>::reverse_iterator>(toRemove.rbegin(), toRemove.rend()))
+            //{
+            //    m_groupings.erase(m_groupings.begin() + index);
+            //}
+
+            // This function is called since it clears all data. A few cpu cycles are lost to settings the m_cellsize to itself.
```
Comment (Algo-ryth-mix) [inline legion/engine/physics/broadphasecollisionalgorithms/broadphaseuniformgrid.cpp]: please remove the dead code

File: legion/engine/physics/systems/physicssystem.hpp [hunk 2]
```diff
@@ -45,6 +45,7 @@ namespace legion::physics
         void fixedUpdate(time::time_span<fast_time> deltaTime)
         {
             static time::timer physicsTimer;
+            log::debug("------------------------------------------------");
```
Comment (kukash) [inline legion/engine/physics/systems/physicssystem.hpp]: probably remove these logs before merging

File: legion/engine/physics/systems/physicssystem.hpp [hunk 3]
```diff
@@ -223,8 +229,8 @@ namespace legion::physics
                         }
                     }
                 }
-                log::debug("groupings {}", manifoldPrecursorGrouping.size());
-                log::debug("total checks {}", totalChecks);
```
Comment (kukash) [inline legion/engine/physics/systems/physicssystem.hpp]: same here,probably you should remove logs

**Review by kukash — APPROVED** [review]

File: legion/engine/physics/systems/physicssystem.hpp (hunk 2 above)
Comment (Algo-ryth-mix) [inline legion/engine/physics/systems/physicssystem.hpp]: please remove the debugging code

File: legion/engine/physics/systems/physicssystem.hpp [hunk 4]
```diff
@@ -223,8 +229,8 @@ namespace legion::physics
...
+                //log::debug("groupings {}", manifoldPrecursorGrouping.size());
+                //log::debug("total checks {}", totalChecks);
```
Comment (Algo-ryth-mix) [inline legion/engine/physics/systems/physicssystem.hpp]: please actually remove the debug code

**Review by Algo-ryth-mix — CHANGES_REQUESTED** [review]

**Review by Algo-ryth-mix — APPROVED** [review]

**Pull request merged**